parasolr/query.py

```diff
@@ -159,13 +160,68 @@ def get_facets(self) -> Dict[str, int]:
         return self.solr.query(**query_opts).facet_counts['facet_fields']
 
     @staticmethod
-    def _lookup_to_filter(key, value) -> str:
-        """Convert keyword argument key=value pair into a Solr filter.
-        Currently only supports simple case of field:value."""
+    def _lookup_to_filter(key: str, value: Any) -> str:
+        """Convert keyword/value argument, with optional filters, into a
+        Solr query. Currently supports simple key-value pairs, searching for
+        empty/unset values, and __in filters of lists.
+
+            Returns: A propertly formatted Solr query string.
+        """
+
+        # Flag if a null/empty search is being conducted
+        null_filter = False
+
+        # Base filter for all queries
+        _filter = ''
+
+        # Implementations of Django-style filters such as __in=[a, b, c]
+        # or __range=(start, end).
+
+        # Parse these special cases first and handle simple key/value pair
+        # at the very end.
+
+        # Regex to split for a double-underscore at end of a filter
+        # since Solr fields, in theory, could have double-underscores in name
+        double_underscore_regex = re.compile(r'_{2}(?=[A-Za-z]+$)')
+
+        # value is a list, join with OR logic for all values in list,
+        # include the empty value for a missing/empty field
+        if key.endswith('__in'):
+            # filter out a negative search and set flag
+            null_filter = bool([item for item in value if not item])
+
+            value = [item for item in value if item]
+            key = re.split(double_underscore_regex, key)[0]
+            if value:
+                _filter = '(%s)' % ' OR '.join(value)
+
+        # dictionary for easier interpolation of values
+        filter_vals = {
+            'filter': _filter if _filter else value,
+            'key': key
+        }
+
+        # if filter_vals['filter] can be cast to int,
+        # do so for comparison against 0 since we don't want to treat
+        # that value as implicitly falsy
+        try:
+            filter_vals['filter'] = int(filter_vals['filter'])
+        except ValueError:
+            pass
+
+        if not filter_vals['filter'] and filter_vals['filter'] != 0:
+            null_filter = True
+
+        # there is a key:value pair and a search for a null via an empty string
+        # or None
+        if null_filter and value:
```

Comment:
Agree on explicitly splitting this out. I will have to handle `''` in lists though, because that causes issues for S&Co's case otherwise. (Alternately, we *could* explicitly index `Unknown` and not use the missing functionality and that simplifies the case significantly.)